DataFormats/Common/interface/DetSetNew.h

```diff
@@ -75,6 +75,9 @@ namespace edmNew {
     inline
     const_iterator end() const { return data()+m_size;}
 
+   DataContainer const & container() const { return *m_data; }
```

Comment:
I'm afraid I don't like exposing the internal details of how the container works. It appears that you did this to allow a different ```makeRefTo```.  I believe you could accomplish the same goal by changing your new ```makeRefTo``` to have as a third argument the ```DetSet``` instead of index one must know how to properly calculate.  Then one could just make that ```makeRefTo``` be a friend of ```DetSet``` and access the container.